src/traces/heatmap/clean_2d_array.js

```diff
@@ -9,16 +9,17 @@
 'use strict';
 
 var isNumeric = require('fast-isnumeric');
+var Lib = require('../../lib');
 
-module.exports = function clean2dArray(zOld, transpose) {
+module.exports = function clean2dArray(zOld, trace, xa, ya) {
     var rowlen, collen, getCollen, old2new, i, j;
 
     function cleanZvalue(v) {
         if(!isNumeric(v)) return undefined;
         return +v;
     }
 
-    if(transpose) {
+    if(trace && trace.transpose) {
```

Comment:
Similarly for `trace`. Is there a situation where `clean2dArray` w/o `trace`?

Reply:
> Similarly for `trace`. Is there a situation where `clean2dArray` w/o `trace`?

Ok, probably not :sweat_smile: . Reverted in ebf545ff33bd2f9fee38b516d73c891a2a2d1bee

Reply:
Turns out this was necessary:
![Screenshot_2019-05-07_11-50-53](https://user-images.githubusercontent.com/301546/57313853-6b6e5200-70be-11e9-8b2c-3b21148f8dab.png)